src/FSharp.Core/reflect.fs

```diff
@@ -266,108 +266,103 @@ module internal Impl =
     //-----------------------------------------------------------------
     // ATTRIBUTE DECOMPILATION
 
-    let tryFindCompilationMappingAttribute (attrs: obj array) =
+    let findCompilationMappingAttributeAllowMultiple (attrs: obj array) =
         match attrs with
-        | null
-        | [||] -> None
-        | [| res |] ->
-            let a = (res :?> CompilationMappingAttribute)
-            Some(a.SourceConstructFlags, a.SequenceNumber, a.VariantNumber)
-        | _ -> invalidOp (SR.GetString(SR.multipleCompilationMappings))
```

Comment:
This is the breaking change for already shipped fslib.
Right now, if they encounter struct DU with overlaps => they report incomplete data.
After this PR, they will deliberately crash with `multipleCompilationMappings` instead of being wrong.


That only affects old fslib encountering a field-overlap-struct DU compiled with fresh SDK (with this PR in it).